reinforcement_learning/agent/rl_agent.py

```diff
@@ -0,0 +1,144 @@
+"""MovieBot agent for reinforcement learning of dialogue policy."""
+import logging
+from typing import Any, Dict, List
+
+from dialoguekit.core import AnnotatedUtterance, Intent
+from dialoguekit.participant import DialogueParticipant
+
+from moviebot.agent.agent import MovieBotAgent
+from moviebot.core.core_types import DialogueOptions
+from moviebot.core.utterance.utterance import UserUtterance
+from moviebot.dialogue_manager.dialogue_act import DialogueAct
+from moviebot.nlu.neural_nlu import NeuralNLU
+from reinforcement_learning.agent.rl_dialogue_manager import DialogueManagerRL
+
+logger = logging.getLogger(__name__)
+
+
+class MovieBotAgentRL(MovieBotAgent):
+    def __init__(self, config: Dict[str, Any] = None) -> None:
+        """Initializes a MovieBot agent for reinforcement learning.
+
+        Args:
+            config: Configuration. Defaults to None.
+        """
+        super().__init__(
+            config=config,
+        )
+
+        self.dialogue_manager = DialogueManagerRL(
+            self.data_config, self.isBot, self.new_user
+        )
+
+        if config.get("nlu_type", "") == "neural":
+            self.nlu = NeuralNLU(None)
+
+    def initialize(self) -> None:
+        """Initializes the agent."""
+        self.dialogue_manager.initialize()
+
+    def generate_utterance(
+        self,
+        agent_dacts: List[DialogueAct],
+        options: DialogueOptions = {},
+        user_fname: str = None,
```

Comment:
Why is `user_fname` added here? I presume this is needed for NLG to output custom text. If this is the case, It should be a part of agent DialogueAct (HI), maybe retrieved from UserModel?

Reply:
As assumed, `user_fname` is needed for NLG. While you suggestion to add it as port of the diaolgue act HI, it should be part of another PR related to the improvement of NLG.